### ISSUE TITLE
Question about SDL2 resolution setting

### ISSUE THREAD

**Issue by christianhaitian** [jtothebell/fake-08#165]
Hi,

First off, great work with this pico-8 player.  It works very well for many pico-8 carts with minor issues.  It's a nice alternative to the official pico-8 player.  With the SDL2 make option, the resolution is fixed in the code instead of using SDL2 to get the current display resolution.  Is there a particular reason for that?  I ask because I implemented this example (https://wiki.libsdl.org/SDL_GetCurrentDisplayMode#code_examples) to set the resolution per the current display and it seems to work fine in my limited testing.  If no particular reason, any concern if I submit a PR with that change for consideration?

Thanks,

**Comment by jtothebell** [comment]
The only reason for the hard coded resolution was to test different stretch modes at the resolution of the Vita and Switch. The SDL 1.2 and SDL 2 builds were really only intended to make debugging issues easier and not intended for distribution, but that was some time ago and the platforms I'm building for has grown beyond just console homebrew with fixed resolutions. I still don't really want to distribute builds for desktop OS's, but I'd absolutely consider a PR using the current display resolution.

Out of curiosity, are you building it for use with ArkOS or another handheld platform?

**Comment by christianhaitian** [comment]
I’m building for ArkOS.

**Comment by jtothebell** [comment]
Ok, good to know. I've shy'ed away from building Fake-08 for any platforms that can run Pico-8 proper to encourage people to use the real thing when possible, but it could make sense to include in ArkOS. You're probably already planning on updating the docs (sidenote: the wikis for ArkOS and Portmaster are great), but I'd just ask that you make it clear that Fake-08 is not affiliated with Lexaloffle, and is not 100% compatible, nor does it have SPLORE functionality, so using Pico-8 proper is going to be preferable in just about every case.

One other thing to be aware of is that I have a libertro core working in a branch. I haven't had as much time to work on it the last couple weeks, but I think it is getting pretty close to being ready, and it may also make sense to use with platforms like ArkOS where so much of the other functionality uses retroarch.

**Comment by christianhaitian** [comment]
Yes I'd be updating the docs about the inclusion of this emulator and I will include the notes about fake-08 as you stated above.  I went back and forth about whether I should I include this emulator in ArkOS or not as I believe the minor 1 time cost for the official pico-8 player is well worth it for the available content but decided to include it because there are a few who just may not be able to pay that 1 time cost.  So having an alternative limited solution seems to be ok.  I'm hoping it may encourage those who's been on the fence about pico-8 on these portable devices may decide to make the small investment in order to help keep the pico-8 community moving forward.  

Thank you for the kind words about the doc.  It's been great keeping it up to date with feedback from community members and seems to have provided a good set of beginning docs for other projects which has been an awesome experience.

I saw that there was work going to bring this to retroarch which I look forward to including it as well in the central repo used by ArkOS, RetroOZ and other similar projects.

I'll submit a PR for the SDL2 resolution changes shortly.  Thanks again for this great emulator.
